cmd/podman/shared/create.go

```diff
@@ -450,6 +451,12 @@ func ParseCreateOpts(ctx context.Context, c *GenericCLIResults, runtime *libpod.
 
 	// USER
 	user := c.String("user")
+	if usernsMode.IsKeepID() {
+		if user != "" {
+			return nil, errors.Errorf("cannot specify --user with --userns=keep-id")
```

Comment:
Is this really necessary?

Currently the `toolbox` script uses `podman create --user root:root --userns=keep-id ...`. The `--user root:root` is for the container's entry point process, which calls `useradd`, sets up the symlinks for `/etc/resolv.conf`, etc. and needs to run as `root` inside the container.. The `--userns=keep-id` is used so that an interactive shell spawned via `podman exec` runs as the same UID as the host.

Reply:
hm no, I will relax this limitation if it is causing problems to toolbox

Reply:
can you check the new version?

Reply:
Yes, it's fine now. Thanks.